workos/mfa.py

```diff
@@ -78,18 +76,7 @@ def enroll_factor(
             "phone_number": phone_number,
         }
 
-        if type is None:
-            raise ValueError("Incomplete arguments. Need to specify a type of factor")
-
-        if type not in ["sms", "totp"]:
-            raise ValueError("Type parameter must be either 'sms' or 'totp'")
-
-        if (
-            type == "totp"
-            and totp_issuer is None
-            or type == "totp"
-            and totp_user is None
-        ):
+        if type == "totp" and (totp_issuer is None or totp_user is None):
```

Comment:
@PaulAsjes food for thought: it's hard to enforce these sorts of `if 'totp' then require `issuer` and `user` constraints clearly in a rest API. But we have more options in an SDK.

If we were rewriting this SDK from scratch, would it be better to have separate functions `enroll_sms` and `enroll_totp` with different required params?

Reply:
+1 to this. I thought about having to separate methods, or better yet overloaded methods.